Lib/warnings.py

```diff
@@ -172,10 +178,12 @@ def simplefilter(action, category=Warning, lineno=0, append=False):
     'lineno' -- an integer line number, 0 matches all warnings
     'append' -- if true, append to the list of filters
     """
-    assert action in ("error", "ignore", "always", "default", "module",
-                      "once"), "invalid action: %r" % (action,)
-    assert isinstance(lineno, int) and lineno >= 0, \
-           "lineno must be an int >= 0"
+    if action not in ("error", "ignore", "always", "default", "module", "once"):
```

Comment:
What about the `category` argument?

Also it may be worth to move the duplicated code into a subfunction.

Reply:
The category argument is a bit different, it's supposed to be a subclass of `Warning` and nothing else but in some cases like https://github.com/python/cpython/blob/master/Lib/unittest/test/test_case.py#L1435 it will get called with a tuple.

Should we relax the condition on category so it accepts tuples of `Warning` subclasses too or fix `unittest` so it calls simplefilter() multiple time when given a iterable?